### ISSUE TITLE
Make mksquashfs binary location (optionally) configurable

### ISSUE THREAD

**Issue by probonopd** [AppImage/AppImageKit#612]
@adrianschroeter [asked](https://github.com/electron-userland/electron-builder/issues/1478#issuecomment-349911382):

> It would be nice if the mksquashfs binary location could become a configure/compile time option.

**Comment by TheAssassin** [comment]
Maybe ask back whether a CMake option would suffice?

**Comment by probonopd** [comment]
Since he mentioned "compile time option" I think yes.

**Comment by TheAssassin** [comment]
I created a PR in #618. @adrianschroeter this allows for setting the path like `-DMKSQUASHFS_DIR=share/appimagekit`, relative to the install root directory (i.e., the directory which is prepended by `CMAKE_INSTALL_PREFIX` normally. If set, appimagetool will look for mksquashfs relative to the appimagetool binary.

@probonopd we might think about making the directory openSUSE uses the default, though. It isn't any more difficult than putting it in `bin`, and could be considered a "sane default" for other distros such as Nitrux, too.

**Comment by TheAssassin** [comment]
@adrianschroeter this, by the way, should eliminate any maintainer patches, as the binary is also installed into the right directory, this doesn't just alter the "search path" for it.

**Comment by probonopd** [comment]
Actually `/usr/share/...` is for architecture-independent code only.

http://www.pathname.com/fhs/pub/fhs-2.3.html#USRSHAREARCHITECTUREINDEPENDENTDATA

> This hierarchy is intended to be shareable among all architecture platforms of a given OS; thus, for example, a site with i386, Alpha, and PPC platforms might maintain a single /usr/share directory that is centrally-mounted.

So we should put it somewhere else.

**Comment by probonopd** [comment]
`/usr/lib/appimagekit` might possibly be a better idea. Some distributions use `libexec` but I can't find that in FHS.

**Comment by TheAssassin** [comment]
Yes, you're right, it should be `lib/appimagekit`. Edited the comment.

**Comment by TheAssassin** [comment]
Closed via #618.

**Comment by probonopd** [comment]
We are now getting

```
No such file or directory: /usr/local/bin/../lib/appimagekit/mksquashfs
```

Reference:
https://travis-ci.org/probonopd/linuxdeployqt/builds/328445959#L1588

**Comment by TheAssassin** [comment]
This is not an issue in AppImageKit, but a linuxdeployqt one. uploadtool should be blamed, as it deleted the release although it couldn't upload files for a new one.
